db_autorestore.md

```diff
@@ -0,0 +1,89 @@
+    - update the latest snapshot id parameters in AWS SSM Parameter Store 
+ * the second one, *db-restore-from-snapshot*, will mainly focus on triggering the update and restore process:
+    - will copy the latest snapshot ID value to the current snapshot ID and then trigger the main pipeline to deploy the changes and restore the databasess from the latest snapshots
+I took advantage of the already existing AWS Lambda functions that are sharing the database snapshots from one account to the other. The sharing part is not in the scope of this article but it was ilustrated for a better view of the solution.
+
+In AWS SSM Parameter Store, I have created 5 parameters:
+
+ * aurora_current_snapshot_id
+ * aurora_latest_snapshot_id
+ * documentdb_current_snapshot_id 
+ * documentdb_latest_snapshot_id
+ * main_cicd_name
+
+The current value will be the one from which the databases are restored from and the latest value will be the value of the most recent snapshot that exists in the account for DocumentDB or Aurora.
+We need to have 2 different parameters, one for the current and one for the latest snapshot, to be able to restore the databases from a specific snapshot ID outside of this process.
+There are 2 pipeline in the account, so the name of the main pipeline, the one that is deploying the infrastructure, was needed.
+
+The entire solution was written using IaC in AWS CDK with Python.
+
+![Architecture](db_autorestore)
+
+## How does it work
+Using a cronjob the update function (*db-update-latest-snapshot-id*) will run every 2 weeks at 2 AM.
+This function has the following tasks/functions:
+
+* *copy_docdb_snapshot* - checks for the latest DocumentDB shared and manual snapshot, and it takes the latest value for both of them. It then checks if the latest shared snapshot is already copied and if not, it will copy that one, if yes, it will return a nice message. 
+
+* *get_latest_rds_snapshots* - checks for the latest shared snapshot for Aurora DB and it gets the latest one.
+
+* *update_ssm* - it will update the SSM Parameter store with the new values of the latest snapshots in *documentdb_latest_snapshot_id* and in *aurora_latest_snapshot_id*.
+
+And this is it for the update process.
+Now for the restore is even simpler. Using another cronjob the restore function  (*db-restore-from-snapshot*) will run every 2 weeks, 2 hours later than the update process.
+The tasks of this functions are:
+
+* *update_current* - it gets the value from the latest parameters and copies it to the current one. Now the value of the *documentdb_current_snapshot_id* will be updated with the value of *documentdb_latest_snapshot_id*. The pipeline will check the value of the current parameters. We have split this process with latest and current parameters to facilitate the restore from a snapshot in other cases than the ones described in this article as menioned above.
+
+* trigger the deployment pipeline using the *main_cicd_name* from the SSM Parameter Store
```

Comment:
Maybe a small code snippet from the CDK using the SSM Parameter could clarify a bit more what the pipeline does?